### PR TITLE
WIP: migrate demo to browserify

### PR DESCRIPTION
**Pull request by xjamundx**
A pretty straightforward port of the current react attempt to use browserify.

To build an updated version of the demo:
`npm run build:eslint`

Or for rapid testing I'm using:
```
watchify --extension=.jsx --debug -x eslint -x espree -t babelify js/app/demo/main.js -o js/app/demo.js
```

Things we could continue to improve:
1. Move react and react dom into `package.json` as `devDependencies` and out of the package.json `browser` field
2. Try out redux or whatever
3. Use more ES6 now, because it's possible :)
2. Worry about minifying or other optimizations

Hardest part of this was manually hacking the orion build to support JS syntax highlighting by default.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,9 +22,16 @@
   "bugs": {
     "url": "https://github.com/eslint/eslint.github.io/issues"
   },
+  "browser": {
```
Comment (xjamundx) [inline package.json]: We can move this stuff into the dev dependencies and remove them from here. I just kept the old stuff, because it worked :)

**Comment by ilyavolodin** [comment]
Hmm.. I though I posted a comment earlier, but it's not showing up.
Just so that I understand, other then `setTimeout` around lint process, the only other change in this PR is requirejs -> browserify + babel + es6, right?

File: js/app/demo/app.jsx [hunk 2]
```diff
@@ -1,88 +1,96 @@
...
-                        atomtest: false,
-                        embertest: false,
-                        webextensions: false,
-                        es6: false,
-                        greasemonkey: false
+// injected via <script>
+const eslint = window.eslint;
+
+const App = React.createClass({
+    getInitialState: function() {
```
Comment (hzoo) [inline js/app/demo/app.jsx]: Maybe add eslint and turn on the `object-shorthand` rule 😄

**Comment by xjamundx** [comment]
Yeah @ilyavolodin nothing cool in here yet, just straight porting your stuff

**Comment by xjamundx** [comment]
Mest around with cleaning up `rulesConfig.jsx` and I think I made it a little cleaner to read and want to share some minor perf. updates.

Here's the performance of toggle one of the **environments** checkboxes (not modified):
screen shot 2016-04-25 at 8 57 35 am

Here's the performance of toggling one of the **rules** checkboxes:
screen shot 2016-04-25 at 8 57 15 am

For the record an uncontrolled checkbox looks like it takes around 10ms to render the changes, so just using react to render it does add around 50ms. More work coming soon!

/ @ilyavolodin  ;-)

File: js/app/demo/rulesConfig.jsx [hunk 3]
```diff
@@ -50,7 +46,6 @@ const RulesConfig = React.createClass({
             change[key][0] = e.target.checked ? 'error' : 'off';
         }
         this.setState(change);
-//      this.props.onUpdate(this.state)
```
Comment (xjamundx) [inline js/app/demo/rulesConfig.jsx]: Ultimately this is what's slowing us down, the rest of the stuff I think had little effect (still testing). I'm going to try running this async or a few other things...

**Comment by xjamundx** [comment]
Here's an interesting approach in `environments.jsx`:
```
this.setState(environment, function() {
     setTimeout(() => this.props.onUpdate(this.state), 100);
});
```

Look at the perf graph:
screen shot 2016-04-25 at 9 07 11 am

So you can see pretty clearly the `setState` isn't slowing us down here at all, it's just all the `onUpdate` call.

**Comment by ilyavolodin** [comment]
@xjamundx I still don't feel comfortable with build step, sorry. I know it's normal to use transpilers, but I really dislike them, and I don't think ES6 really buys us anything in this case. I'm going to close this for now (but will leave the branch around, just in case).

**Comment by xjamundx** [comment]
No worries :)

**Pull request closed without merging**